app/controllers/status_controller.rb

```diff
@@ -24,7 +24,7 @@ def ping
 private
 
   def database_alive?
-    ActiveRecord::Base.connection.active?
+    ActiveRecord::Base.connection.verify!
```

Comment:
I've been doing some digging and there is an alternative
```ruby
ActiveRecord::Base.connection.database_exists?
```
Under the hood, [this runs a connect! on the database](https://api.rubyonrails.org/classes/ActiveRecord/ConnectionAdapters/AbstractAdapter.html#method-i-database_exists-3F) that fails if the connect drops.
```ruby
def database_exists?
  connect!
  true
rescue ActiveRecord::NoDatabaseError
  false
end
```

Reply:
I guess this is preferable because it captures the relevant error internally so we don't have to concern ourselves with it what that might be. The current mechanism relies on the raising of a specific PG error too, which I am not sure is raised by `verify!` under the hood. But the `verify!` does use `active?` and [only connects if active is false](https://api.rubyonrails.org/classes/ActiveRecord/ConnectionAdapters/AbstractAdapter.html#method-i-verify-21) so is "lazier".


Anyway overall `database_exists?` seems safer and more reliable so will switch that out